### DISCUSSION TITLE
Battery sensor

### DISCUSSION THREAD

**Discussion by Hjalti14** [alandtse/tesla#46] [Q&A] [answered]
Hello all

I just added the tesla custom intergration. Why do I not see the battery sensor? I can only see these entities:

Tesla Model Y charger door lock
lock.tesla_model_y_charger_door_lock

Tesla Model Y charger sensor
binary_sensor.tesla_model_y_charger_sensor

Tesla Model Y charger switch
switch.tesla_model_y_charger_switch

Tesla Model Y door lock
lock.tesla_model_y_door_lock

Tesla Model Y frunk lock
lock.tesla_model_y_frunk_lock

Tesla Model Y HVAC (climate) system
climate.tesla_model_y_hvac_climate_system

Tesla Model Y location tracker
device_tracker.tesla_model_y_location_tracker

Tesla Model Y maxrange switch
switch.tesla_model_y_maxrange_switch

Tesla Model Y online sensor
binary_sensor.tesla_model_y_online_sensor

Tesla Model Y parking brake sensor
binary_sensor.tesla_model_y_parking_brake_sensor

Tesla Model Y polling switch
switch.tesla_model_y_polling_switch

Tesla Model Y sentry mode switch
switch.tesla_model_y_sentry_mode_switch

**Comment by omelhus** [comment]
This should probably be filed under issues, preferably using the issue template.

Would you be so kind to provide us with some logs, versions etc? It’s impossible to guess based on what you’ve provided here.

**Comment by Hjalti14** [comment]
Thank for the help mate

These are the logs that I get:

**Logger: homeassistant.setup
Source: setup.py:310
First occurred: 11:39:41 (1 occurrences)
Last logged: 11:39:41

Unable to prepare setup for platform tesla_custom.sensor: Platform not found (cannot import name 'STATE_CLASSES' from 'homeassistant.components.sensor' (/srv/homeassistant/lib/python3.8/site-packages/homeassistant/components/sensor/__init__.py)).**

The version I am using is Home Assistant 2021.5.4

**Reply by omelhus** [reply]
You need to upgrade HA to make this work. There was a change mid may that the integration requires in order to work.

https://github.com/home-assistant/core/pull/50063

**Comment by alandtse** [comment, marked as answer]
Also, this was addressed in an issue. https://github.com/alandtse/tesla/issues/19
